### DISCUSSION TITLE
HTML-centric building pipeline

### DISCUSSION THREAD

**Discussion by textpain** [parcel-bundler/parcel#6101] [Q&A]
Hello everyone.

I often build static websites that are ultimately centered around HTML and CSS, not JavaScript. So it made me think if I can use `index.html` as an entry point and base my entire building system around it.

The idea is to have HTML files that can import scripts, styles, assets and other HTML files that can import scripts, styles, assets and so on, entirely recursively just like what Parcel does with JS. As a build result I want to see one HTML file, one JS file and one CSS file. 

Parcel works perfectly with everything JS. I tried PostHTML and literally every plugin that comes with it and mentions imports or CSS. The best I could achieve was importing HTML files into one another, but that's what I can do with plain concat, not a full-blown HTML parser. That's a bummer because both PostHTML and PostCSS build AST which is more than enough for accomplishing what I need with a little help from CSS modules.

The idea seems perfectly clear and desirable (several of my friends struggled with similar tasks) and it makes me wonder if it was already implemented and it was me who missed it.

Suggestions welcome. Thanks in advance.

**Comment by mischnic** [comment]
So you essentially want to use https://github.com/posthtml/posthtml-include with Parcel?

This should work simply as a PostHTML plugin: https://v2.parceljs.org/languages/html/

**Reply by textpain** [reply]
I tried it and I was unable to make relative imports work. Also, I couldn't make it work with some kind of CSS bundling (my CSS imports was inside `body` at best, just like with just concatenating HTML files into one another)
